scripts/plugins/extauth-hook-AD.py

```diff
@@ -205,18 +244,19 @@ def is_special_line(self, line):
 
     def _load_values(self):
```

Comment:
I understand this is a packport but I notice a deeply nested logic that I don't trust to be correct. There are several special cases (empty lines, kv != 2) that are mostly unmotivated. Given that this code handles part of authentication I would like to see more clarity.